### ISSUE TITLE
IntelliJ IDEA requires modules for langchain4j

### ISSUE THREAD

**Issue by InAnYan** [InAnYan/jabref#6]
### JabRef version

5.13 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 11

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

Just write the code with `langchain4j` library.

### Appendix

The error IntelliJ gives is:
```
Package 'dev.langchain4j.data.message' is declared in module 'langchain4j.core', but module 'org.jabref' does not read it
```
The same applies for other classes.

However, the code compiles without any errors.

Moreover, if we apply the IDEA's advice, the building breaks.

**Comment by InAnYan** [comment]
@koppor In the repository [intellij-issue-modularity](https://github.com/InAnYan/intellij-issue-modularity) I made all the changes, but it stuck at error:
```
D:\Git\intellij-issue-modularity\src\main\java\org\example\Main.java:5: error: cannot find symbol
import dev.langchain4j.model.embedding.AllMiniLmL6V2EmbeddingModel;
                                      ^
  symbol:   class AllMiniLmL6V2EmbeddingModel
  location: package dev.langchain4j.model.embedding
```

Could you please update the repo with the changes you made in JabRef?

**Comment by koppor** [comment]
Can't we add `--patch-module` (based on the contents of `build.gradle`) similar to the other "Compilation options" given at https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-12-build.html.

**Comment by koppor** [comment]
> Could you please update the repo with the changes you made in JabRef?

Done. See https://github.com/InAnYan/intellij-issue-modularity/commit/7ba30faf55c0a4ebf14a83beb1345c624a632fca.

**Comment by koppor** [comment]
BTW, there is a work-around:

Please add following things to `intellij-12-build.md` (rendered at https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-12-build.html)

```
--add-reads
org.example=langchain4j.embeddings.all.minilm.l6.v2
--add-reads org.example=langchain4j.core
```

In the text box where other paramters to `JabRef.main` module are added:

![image](https://github.com/InAnYan/jabref/assets/1366654/ff183f5a-e48c-4f75-a1db-994baafeb342)

**Comment by InAnYan** [comment]
New parameters:
```
--add-exports=javafx.controls/com.sun.javafx.scene.control=org.jabref
--add-exports=org.controlsfx.controls/impl.org.controlsfx.skin=org.jabref
--add-reads
org.jabref=org.fxmisc.flowless
--add-reads
org.jabref=org.apache.commons.csv
--add-reads
org.jabref=langchain4j.embeddings.all.minilm.l6.v2
--add-reads
org.jabref=langchain4j.embeddings.all.minilm.l6.v2.q
--add-reads
org.jabref=langchain4j.core
--add-reads
org.jabref=langchain4j.open.ai
```

**Comment by koppor** [comment]
With the fix, there are no compiler errors, the imports are without errors, too:

![image](https://github.com/InAnYan/jabref/assets/1366654/daaab9e2-3ded-45fa-8bdb-55671f9a6f81)

Thus, closing this issue.

**Comment by koppor** [comment]
Reopening; cannot get `--patch-module` to work. I always get

![image](https://github.com/InAnYan/jabref/assets/1366654/068dbbde-7e78-4269-9588-a95b48d6be8b)

---

Blocked by https://github.com/langchain4j/langchain4j/issues/1066

**Comment by koppor** [comment]
@InAnYan This can be closed, because with the configuration of the compiler, the issue indicated by the title is working

@InAnYan I would propose to open another issue for compilation with IntelliJ referencing to https://github.com/langchain4j/langchain4j/issues/1066. This is different enough to this one.
